corehq/apps/consumer_user/tests/test_views.py

```diff
@@ -0,0 +1,333 @@
+                                                           email=email)
+        User.objects.create_user(username=email, email=email, password='password')
+        register_uri = register_url(invitation.id)
+        response = self.client.get(register_uri)
+        self.assertEqual(response.status_code, 302)
+        self.assertTemplateNotUsed(response, 'signup.html')
+        self.assertFalse(invitation.accepted)
+
+    def test_register_invalid_invitation(self):
+        register_uri = register_url(1000)
+        response = self.client.get(register_uri)
+        self.assertEqual(response.status_code, 400)
+        self.assertTemplateNotUsed(response, 'signup.html')
+
+
+class LoginTestCase(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.login_url = reverse('consumer_user:patient_login')
+        self.homepage_url = reverse('consumer_user:patient_homepage')
+        self.client = Client()
+
+    def test_login_get(self):
+        response = self.client.get(self.login_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'p_login.html')
+
+    def test_login_post(self):
+        email = 'log@log.in'
+        password = 'password'
+        user = User.objects.create_user(username=email, email=email, password=password)
+        ConsumerUser.objects.create(user=user)
+        post_data = {
+            'auth-username': email,
+            'auth-password': password,
+            'patient_login_view-current_step': 'auth',
+        }
+        response = self.client.post(self.login_url, post_data)
+        self.assertEqual(response.status_code, 302)
+        self.assertRedirects(response, self.homepage_url)
+
+    def test_login_accept_invitation(self):
+        email = 'log1@log1.in'
+        password = 'password'
+        user = User.objects.create_user(username=email, email=email, password=password)
+        ConsumerUser.objects.create(user=user)
+        invitation = ConsumerUserInvitation.objects.create(case_id='6', domain='1', invited_by='I',
+                                                           invited_on='2021-02-09 17:17:32.229524+00',
+                                                           email=email)
+        post_data = {
+            'auth-username': email,
+            'auth-password': password,
+            'patient_login_view-current_step': 'auth',
+        }
+        response = self.client.post(login_accept_url(invitation.id), post_data)
+        self.assertEqual(response.status_code, 302)
+        invitation.refresh_from_db()
+        self.assertTrue(invitation.accepted)
+        self.assertRedirects(response, self.homepage_url)
+
+
+class SignalTestCase(TestCase):
+
+    def setUp(self):
+        self.domain = 'consumer-invitation-test'
+
+    @override_settings(TESTS_SHOULD_USE_SQL_BACKEND=True)
+    def test_method_send_email(self):
+        invitation_count = ConsumerUserInvitation.objects.count()
```

Comment:
You need to delete all invitations after each test (or  test class) with a teardown. You shouldn't need this line.

Reply:
Added teardown to all testclasses and deleting all objects in it.